File: Pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Get input from user
    string text = get_string("Text: ");
    float L = (count_letters(text) / (float) count_words(text)) * 100;
    float S = (count_sentences(text) / (float) count_words(text)) * 100;
    // Compute Coleman-Liau index
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if(index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
    // To check individual functions: printf("%i letters\n %i words\n %i sentences\n", count_letters(text), count_words(text), count_sentences(text));
}

// Count no. of letters
int count_letters(string text)
{
    int num_letter = 0;

    for(int i = 0; i < strlen(text); i++)
    {
        if(isalpha(text[i]))
        {
            num_letter++;
        }
    }
    return num_letter;
}

// Count no. of words
int count_words(string text)
{
    int num_words = 0;

    for(int j = 0; j <= strlen(text); j++)
    {
    /* Check condition where first character is not blank space and condition where no adjacent spacing
        Need to include j == 0 as we only want pass this condition once for the loop and we need to include back the first word which is omitted during the second condition
        No need to consider j == len(text)-1 during second condition where last character may be blank space as return from strlen(text) will exclude it */
        if ((j == 0 && text[0] != ' ') || (text[j] == ' ' && text[j + 1] != ' '))
        {
            num_words ++;
        }
        else if(text[j] == '-')
        {

        }
    }
    return num_words;
}

// Count no. of sentences
int count_sentences(string text)
{
    int num_sentences = 0;

    for(int k = 0; k < strlen(text); k++)
    {
        if(text[k] == '.' || text[k] == '!' || text[k] == '?' )
        {
            num_sentences++;
        }
    }
    return num_sentences;
}

File: Pset6/sentimental-readability/readability.py
from cs50 import get_string

# TODO
def main():

    # Get input from user
    text = get_string("Text: ").strip()
    L = count_letters(text) / count_words(text) * 100
    S = count_sentences(text) / count_words(text) * 100
    A = count_words(text)
    # Compute Coleman-Liau index
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


# Count no. of letters
def count_letters(text):
    num_letters = 0

    for i in range(len(text)):
        if text[i].isalpha():
            num_letters += 1

    return num_letters


# Count no. of words
def count_words(text):
    num_words = 0

    for j in range(len(text)):
        # Check condition where first character is not blank space and condition where no adjacent spacing
        # Need to include j == 0 as we only want pass this condition once for the loop and we need to include back the first word which is omitted during the second condition
        # No need to consider j == len(text)-1 during second condition where last character may be blank space as return from len(text) will exclude it
        if (j == 0 and text[j] != ' ') or (text[j] == ' ' and text[j + 1] != ' '):
                num_words += 1
        elif text[j] == '-':
                num_words

    return num_words


# Count no. of sentences
def count_sentences(text):
    num_sentences = 0

    for k in range(len(text)):
        if text[k] == '.' or text[k] == '!' or text[k] == '?':
            num_sentences += 1

    return num_sentences


# Python's way of calling main function after user defined function
if __name__ == "__main__":
    main()
